packages/launchpad/cypress/e2e/integration/choose-a-browser.spec.ts

```diff
@@ -0,0 +1,280 @@
+    cy.get('h1').should('contain', 'Configuration Files')
+    cy.contains('Continue').click()
+    cy.get('h1').should('contain', 'Initializing Config...')
+    cy.contains('Next Step').click()
+  }
+
+  // Mocks server's browser detection with known list
+  const setupMockBrowsers = function () {
+    cy.withCtx(async (ctx, o) => {
+      if (ctx._apis.appApi.getBrowsers.restore) {
+        ctx._apis.appApi.getBrowsers.restore()
+      }
+
+      const mockBrowsers = [{
+        'id': '1',
+        'channel': 'stable',
+        'disabled': false,
+        'isSelected': true,
+        'displayName': 'Chrome',
+        'family': 'chromium',
+        'majorVersion': '1',
+        'name': 'chrome',
+        'path': '/test/chrome/path',
+        'version': '1.2.333.445',
+      } as FoundBrowser, {
+        'id': '2',
+        'channel': 'stable',
+        'disabled': false,
+        'isSelected': false,
+        'displayName': 'Firefox',
+        'family': 'firefox',
+        'majorVersion': '2',
+        'name': 'firefox',
+        'path': '/test/firefox/path',
+        'version': '2.3.444',
+      } as FoundBrowser, {
+        'id': '3',
+        'channel': 'stable',
+        'disabled': false,
+        'isSelected': false,
+        'displayName': 'Electron',
+        'family': 'chromium',
+        'majorVersion': '3',
+        'name': 'electron',
+        'path': '/test/electron/path',
+        'version': '3.4.555.66',
+      } as FoundBrowser, {
+        'id': '4',
+        'channel': 'stable',
+        'disabled': false,
+        'isSelected': false,
+        'displayName': 'Edge',
+        'family': 'chromium',
+        'majorVersion': '4',
+        'name': 'edge',
+        'path': '/test/edge/path',
+        'version': '4.5.666.77',
+      }]
+
+      sinon.stub(ctx._apis.appApi, 'getBrowsers').resolves(mockBrowsers)
+    })
+  }
+
+  // Force the server to return no detected browsers
+  const setupZeroMockBrowsers = function () {
+    cy.withCtx(async (ctx, o) => {
+      if (ctx._apis.appApi.getBrowsers.restore) {
+        ctx._apis.appApi.getBrowsers.restore()
+      }
+
+      sinon.stub(ctx._apis.appApi, 'getBrowsers').resolves([])
+    })
+  }
+
+  // Creates and returns aliases for browsers found on the DOM
+  const getBrowserAliases = function () {
+    cy.get('[data-cy="open-browser-list"] [data-selected-browser]').eq(0).as('chromeRadioOption')
+    cy.get('[data-cy="open-browser-list"] [data-selected-browser]').eq(1).as('firefoxRadioOption')
+    cy.get('[data-cy="open-browser-list"] [data-selected-browser]').eq(2).as('electronRadioOption')
+    cy.get('[data-cy="open-browser-list"] [data-selected-browser]').eq(3).as('edgeRadioOption')
+
+    return {
+      chrome: '@chromeRadioOption',
+      firefox: '@firefoxRadioOption',
+      electron: '@electronRadioOption',
+      edge: '@edgeRadioOption',
+    }
+  }
+
+  beforeEach(() => {
+    cy.scaffoldProject('launchpad')
+  })
+
+  describe('with found browsers', () => {
```

Comment:
total nit, when reading this the through, I wasn't expected the test case `shows warning when launched with --browser name that cannot be found` given this describe block name.

Reply:
That's fair, I'll workshop these outer scopes